server/src/main/java/org/opensearch/wlm/tracker/QueryGroupResourceUsageTrackerService.java

```diff
@@ -8,34 +8,34 @@
 
 package org.opensearch.wlm.tracker;
 
-import org.opensearch.tasks.Task;
 import org.opensearch.tasks.TaskResourceTrackingService;
 import org.opensearch.wlm.QueryGroupLevelResourceUsageView;
 import org.opensearch.wlm.QueryGroupTask;
 import org.opensearch.wlm.ResourceType;
 
-import java.util.EnumMap;
 import java.util.EnumSet;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /**
  * This class tracks resource usage per QueryGroup
  */
 public class QueryGroupResourceUsageTrackerService {
-
     public static final EnumSet<ResourceType> TRACKED_RESOURCES = EnumSet.allOf(ResourceType.class);
     private final TaskResourceTrackingService taskResourceTrackingService;
+    private final Supplier<Long> nanoTimeSupplier;
 
     /**
      * QueryGroupResourceTrackerService constructor
      *
      * @param taskResourceTrackingService Service that helps track resource usage of tasks running on a node.
      */
-    public QueryGroupResourceUsageTrackerService(TaskResourceTrackingService taskResourceTrackingService) {
+    public QueryGroupResourceUsageTrackerService(TaskResourceTrackingService taskResourceTrackingService, Supplier<Long> nanoTimeSupplier) {
```

Comment:
This should go away once we push the `Supplier<Long> nanoTimeSupplier` to one of the constructors

Reply:
Done.